### PR TITLE
Make FakeSequence Iterable

### PR DESCRIPTION
**Pull request by NarekDW**
## Make FakeSequence Iterable

To allow to iterate by fake sequence like this:
```java
FakeSequence<String> digits = faker
        .stream(() -> faker.number().digit())
        .len(10)
        .build();

for (String digit : digits) 
    System.out.println(digit);
```

### CODE REVIEW

File: src/main/java/net/datafaker/sequence/FakeStream.java [hunk 1]
```diff
@@ -25,6 +26,11 @@ public boolean isInfinite() {
         return maxLength < 0;
     }
 
+    @Override
+    public Iterator<T> iterator() {
+        return get().iterator();
+    }
+
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/sequence/FakeStream.java:33]: Probably it's better to make such change on FakeSequence level like
```java
    public abstract <Sequence extends Iterable<T>> Sequence get();

    public final Iterator<T> iterator() {
        return get().iterator();
    }
```
and `FakeStream`, `FakeCollection` keep untouched

Comment (NarekDW) [reply src/main/java/net/datafaker/sequence/FakeStream.java:33]: @snuyanzin we can do that, but in that case we have to use Reflection API, as Stream doesn't extend Iterable interface. (It has it's own method with the same name...)
Something like this:
```java
    @Override
    public Iterator<T> iterator() {
        try {
            Object iterable = get();
            Method iterator = iterable.getClass().getMethod("iterator");
            iterator.setAccessible(true);
            //noinspection unchecked
            return (Iterator<T>) iterator.invoke(iterable);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
```

Is this fine?

Comment (snuyanzin) [reply src/main/java/net/datafaker/sequence/FakeStream.java:33]: true, ok then let's go with the initial version

Comment (NarekDW) [reply src/main/java/net/datafaker/sequence/FakeStream.java:33]: Ok, I'm leaving this unchanged

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**